fml/thread.h

```diff
@@ -16,7 +17,52 @@ namespace fml {
 
 class Thread {
  public:
-  explicit Thread(const std::string& name = "");
+  // Valid values for priority of Thread
+  enum class ThreadPriority : int {
+    // Suitable for threads that shouldn't disrupt high priority work.
+    BACKGROUND,
+    // Default priority level.
+    NORMAL,
+    // Suitable for threads which generate data for the display.
+    DISPLAY,
+    // Suitable for thread which raster data
+    RASTER,
+  };
+
+  /// The ThreadConfig is used for setting thread perorities
+  class ThreadConfig {
+   public:
+    explicit ThreadConfig(std::string name = "",
+                          ThreadPriority priority = ThreadPriority::NORMAL);
+
+    static std::unique_ptr<ThreadConfig> DefaultConfigure(
```

Comment:
Functions should have a verb. https://google.github.io/styleguide/cppguide.html#Function_Comments